### PR TITLE
Do not redirect on error

### PR DESCRIPTION
**Pull request by yoshdog**
Currently we redirect when there is an error. Instead lets just post the error response back to the user.

### CODE REVIEW

**Comment by yoshdog** [comment]
```bash
$ npm test                                                                                                                                                                                                                                                                  

> oauth20-provider@0.6.0 test /Users/toan/src/node-oauth20-provider
> mocha --reporter spec



  Authorization Code Grant Type
    ✓ GET /authorization with response_type="code" expect login form redirect
    ✓ POST /login expect authorized (39ms)
    ✓ GET /authorize with response_type="code" expect decision
    ✓ POST /authorize with response_type="code" and decision="1" expect code redirect
    ✓ POST /token with grant_type="authorization_code" expect token
    ✓ POST /secure expect authorized

  Authorization Code Grant Type without client's refresh token grant type
    ✓ GET /authorization with response_type="code" expect login form redirect
    ✓ POST /login expect authorized
    ✓ GET /authorize with response_type="code" expect decision
    ✓ POST /authorize with response_type="code" and decision="1" expect code redirect
    ✓ POST /token with grant_type="authorization_code" expect token but not refresh_token
    ✓ POST /secure expect authorized

  Client Credentials Grant Type
    ✓ POST /token with grant_type="client_credentials" expect token
    ✓ POST /secure expect forbidden
    ✓ POST /client expect authorized

  Emit log event
    ✓ Check log event on POST /token with grant_type="client_credentials" expect token
    ✓ Check a caught exception on POST /token with grant_type="client_credentials" expect token

  Implicit Grant Type
    ✓ GET /authorization with response_type="token" expect login form redirect
    ✓ POST /login authorize
    ✓ GET /authorize with response_type="token" expect decision
    ✓ POST /authorize with response_type="token" and decision="1" expect code redirect
    ✓ POST /secure expect authorized

  Password Grant Type
    ✓ POST /token with grant_type="password" expect token
    ✓ POST /token with grant_type="refresh_token" expect same accessToken
    ✓ POST /secure expect authorized

  Password Grant Type without client's refresh token grant type
    ✓ POST /token with grant_type="password" expect token
    ✓ POST /secure expect authorized

  Refresh Token Grant Type
    ✓ POST /token with grant_type="password" expect token
    ✓ POST /token with grant_type="refresh_token" expect same accessToken
    ✓ Wait and POST /token with grant_type="refresh_token" expect diferent [new]accessToken (2012ms)
    ✓ POST /secure with old token expect forbidden
    ✓ POST /secure witj new token expect authorized


  32 passing (2s)

```

**Comment by benaskins** [comment]
@yoshdog is there a build pipeline for this?

**Comment by benaskins** [comment]
lol, you knew I was going to ask 🥇

**Comment by yoshdog** [comment]
I doubt it... its a forked npm package.

**Comment by benaskins** [comment]
I just wanted to see a green build. Could we also add a unit to test to cover this scenario?

**Comment by yoshdog** [comment]
I wrote some mocha tests a few years ago. let me give it a go... @benaskins

**Comment by runlevel5** [comment]
@yoshdog a test would be good, mocha time

File: test/authorizationCode.js [hunk 1]
```diff
@@ -53,6 +53,18 @@ describe('Authorization Code Grant Type ',function() {
...
+        invalid_client_id = 123
+
+        request(app)
+            .get(authorizationUrl + '?' + query.stringify({client_id: invalid_client_id}))
+            .set('Cookie', cookie)
+            .expect(400, function(err, res) {
+                if (err) return done(err);
+                done();
+            });
+    });
```
Comment (benaskins) [inline test/authorizationCode.js:70]: Does this cover the case where a `redirectUri` is provided?

**Review by benaskins — APPROVED** [review]
LGTM, could you ask @jacobbednarz to look over this as well please?

File: test/authorizationCode.js (hunk 1 above)
Comment (yoshdog) [reply test/authorizationCode.js:70]: does now

File: test/authorizationCode.js [hunk 2]
```diff
@@ -53,6 +53,22 @@ describe('Authorization Code Grant Type ',function() {
             });
     });
 
+    it('GET /authorize with invalid params expect error', function(done) {
```
Comment (jacobbednarz) [inline test/authorizationCode.js:56]: do we need a `POST` case as well?

Comment (yoshdog) [reply test/authorizationCode.js:56]: updated... can you have a look over again @jacobbednarz

**Comment by jacobbednarz** [comment]
looks good 👍

**Pull request merged**